### PR TITLE
ci(): add playwright test suite

### PR DESCRIPTION
**Pull request by ShaMan123**
Added an e2e test using playwright
Added a workflow to execute the test
The workflow produces a [very nice report](https://github.com/deepnest-io/Deepnest/actions/runs/12293315656/artifacts/2310573468)

### CODE REVIEW

File: .github/workflows/playwright.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    steps:
+      - name: boost
+        if: startsWith(matrix.os, 'ubuntu')
+        run: sudo apt-get update && sudo apt-get install -yq libboost-dev
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
+      - name: Install dependencies
+        run: npm ci
+      - name: Build
+        run: npm run build
```
Comment (ShaMan123) [inline .github/workflows/playwright.yml:23]: Apparently the default build step executed as part of npm i/ci wasn't producing the build... That was causing the test to fail

**Review by ShaMan123 — COMMENTED** [review]
This is ready to merge

**Comment by ShaMan123** [comment]
Squash and merge please

**Comment by bosd** [comment]
I've been kicked out of this organization. So maybe someone else can help with the merge.

**Comment by ShaMan123** [comment]
Sounds too much like the org I was kicked out from

**Comment by abebeos** [comment]
> I've been kicked out of this organization. So maybe someone else can help with the merge.

Whats wrong with you, that is getting annoying. You may want to stop spreading misinformation. You're still a member, just without code-access rights.

You have generally a tendency to talk nonsense, and then to keep silence, like e.g. here:

**Comment by abebeos** [comment]
> I actually have merge rights, but I prefer to allow reviews. 

You don't merge your own stuff without a review.

Care to review?

Are you aware that this PR is 11 hours old, and that you pinged already 3 times? Please cool down.

**Comment by ShaMan123** [comment]
@abebeos I did you write that to you but to @bosd 
I wanted to give you the benefit of the doubt after experiencing paternalistic communication which I do not enjoy.
May I remind us all that we are different people of extremely different backgrounds.
Being nice is better than the opposite. I do not want/need to take part in this discussion/org.

**Comment by ShaMan123** [comment]
@abebeos I did you write that to you but to @bosd 
I wanted to give you the benefit of the doubt after experiencing paternalistic communication which I do not enjoy.
May I remind us all that we are different people of extremely different backgrounds.
Being nice is better than the opposite. I do not want/need to take part in this discussion/org.

**Comment by abebeos** [comment]
> @abebeos I did not write that to you but to @bosd

So what? You still pinged me two times.

> I wanted to give you the benefit of the doubt 

How decent of you...

> after experiencing paternalistic communication which I do not enjoy.

"paternalistic" - this is standard-procedure (to protect a work/repo from 'stallion-type' devs which are far too fast and a bit sloppy)

> May I remind us all that we are different people of extremely different backgrounds.

Your background lacks one thing: sticking to standard-procedures.

> Being nice is better than the opposite. I do not want/need to take part in this discussion/org.

Then don't.

> So as before. Take it or leave it. I am done here for good.

Fascinating process.

**Comment by bosd** [comment]
> Whats wrong with you,

I'm really bothered with your toxic communication.

>You have generally a tendency to talk nonsense, and then to keep silence, like e.g. here:

Yes, I'd rather be silent then start an online catfight.
As the mission is to build a organization where people collobarote with respect. Even with different backgrounds.
@ShaMan123 Seems to have the good mindet. Hope he sticks around.

> Being nice is better than the opposite. I do not want/need to take part in this discussion/org.

**Comment by abebeos** [comment]
> I'm really bothered with your toxic communication.

And I'm bothered by your 'toxic' logic and off-topics.

> Yes, I'd rather be silent then start an online catfight.

There's not cat-fight at all - we're not equals.

> As the mission

The mission stopped when payments stopped, and when I decided to continue integration work alone.

That's why I took ownership of this org/code.

Start a new mission if you feel confident enough.

**Comment by abebeos** [comment]
closing, as unmaintained.

**Comment by Dexus** [comment]
Dude, am I stupid for reading all this again? 

@abebeos you may lead the organisation here, but you're not a leader in my eyes. Reading through all this makes me feel really sick. Where do you actually live with such manners? I have to be honest and I've already noticed that you simply delete comments... I don't like that at all. If something goes against your grain, try communicating sensibly. You are not the boss of employees here who you are allowed to treat like "shit", but you have comrades-in-arms here who are actually fighting for the common cause. If you want to stick to this procedure, I'll say directly - I'm out and will gladly accept the help of @ShaMan123 @bosd and whoever else, then let's just set up a new organisation and fork. Where no "dictator" can get lost in macho behaviour. 

Thought I was slowly getting rid of the whole kindergarten....

**Comment by ShaMan123** [comment]
@Dexus you are welcome to ping me on my fork or wherever.
I agree with every word you wrote.
It is a shame that open source becomes so ugly. People forget themselves or think that because they have control over some piece of code or the skill to write it means they are hot shots. 🤢.

**Pull request closed without merging**